### PR TITLE
Fix message parsing for documents containing Content-Length keyword

### PR DESCRIPTION
**Pull request by dinvlad**
## Description

This PR fixes the logic for parsing a batch of messages in `data_received()`, where each message body may contain a literal `Content-Length` keyword.

Previously, if a document contained this keyword, then the server would crash, because it attempted to split messages on this keyword.

Now, instead of splitting the messages on the keyword, it parses content length from the first keyword, and then extracts the message based on the length. After that, the parsing loop is repeated, until no more messages can be read.

Update: I added handling of partial messages, since some messages may arrive in chunks.

## Code review checklist (for code reviewer to complete)

- [x] Pull request represents a single change (i.e. not fixing disparate/unrelated things in a single PR)
- [x] Title summarizes what is changing
- [x] Commit messages are meaningful (see [this][commit messages] for details)
- [x] Tests have been included and/or updated, as appropriate
- [ ] Docstrings have been included and/or updated, as appropriate
- [ ] Standalone docs have been updated accordingly
- [x] [CONTRIBUTORS.md][contributors] was updated, as appropriate
- [x] Changelog has been updated, as needed (see [CHANGELOG.md][changelog])

[changelog]: https://github.com/openlawlibrary/pygls/blob/master/CHANGELOG.md
[commit messages]: https://chris.beams.io/posts/git-commit/
[contributors]: https://github.com/openlawlibrary/pygls/blob/master/CONTRIBUTORS.md

### CODE REVIEW

**Comment by danixeee** [comment]
@dinvlad Thanks for the PR, great work! Is it ready for the review or you plan to make more changes?

**Comment by dinvlad** [comment]
I think it's ready, thanks!

File: CONTRIBUTORS.md [hunk 1]
```diff
@@ -5,3 +5,4 @@
 - [Max O'Cull](https://github.com/Maxattax97)
 - [Tomoya Tanjo](https://github.com/tom-tan)
 - [yorodm](https://github.com/yorodm)
+- [Denis Loginov](https://github.com/dinvlad)
```
Comment (danixeee) [inline CONTRIBUTORS.md]: Can you make this alphabetical?

File: pygls/protocol.py [hunk 2]
```diff
@@ -397,18 +397,43 @@ def connection_made(self, transport: asyncio.Transport):
         """Method from base class, called when connection is established"""
         self.transport = transport
 
+    MESSAGE_PATTERN = re.compile(
```
Comment (danixeee) [inline pygls/protocol.py]: Can you move this up where other class constants are (alphabetically)?

**Comment by dinvlad** [comment]
Awesome, thanks for reviewing! I've made the changes.

**Comment by danixeee** [comment]
I have tested everything and it works. I needed to test _json-extension_ after installing from `.vsix`, since it is using IO connection instead of TCP.

I merged other PR today, can you just resolve conflicts and include link to your PR? 

_CHANGELOG_ should look like this:

```md
- Fix parsing of partial messages and those with Content-Length keyword ([#80])
- Fix Full SyncKind for servers accepting Incremental SyncKind ([#78])

[#80]: https://github.com/openlawlibrary/pygls/pull/80
[#78]: https://github.com/openlawlibrary/pygls/pull/78
```

**Comment by dinvlad** [comment]
Sounds good - resolved, please take a look.

**Pull request merged**